frontend/packages/dev-console/src/components/import/section/BuildOptions.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { SelectVariant } from '@patternfly/react-core/dist/esm/components';
+import { FormikValues, useFormikContext } from 'formik';
+import * as _ from 'lodash';
+import { Trans, useTranslation } from 'react-i18next';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import { useAccessReview } from '@console/internal/components/utils';
+import { CLUSTER_PIPELINE_NS, FLAG_OPENSHIFT_PIPELINE } from '@console/pipelines-plugin/src/const';
+import { PipelineModel } from '@console/pipelines-plugin/src/models';
+import { SelectInputField, SelectInputOption, useFlag } from '@console/shared';
+import { FLAG_OPENSHIFT_BUILDCONFIG, FLAG_SHIPWRIGHT_BUILD } from '../../../const';
+import { BuildOptions, ReadableBuildOptions } from '../import-types';
+
+const usePipelineAccessReview = (): boolean => {
+  const canListPipelines = useAccessReview({
+    group: PipelineModel.apiGroup,
+    resource: PipelineModel.plural,
+    namespace: CLUSTER_PIPELINE_NS,
+    verb: 'list',
+  });
+
+  const canCreatePipelines = useAccessReview({
+    group: PipelineModel.apiGroup,
+    resource: PipelineModel.plural,
+    namespace: getActiveNamespace(),
+    verb: 'create',
+  });
+
+  return canListPipelines && canCreatePipelines;
+};
+
+export const BuildOption = () => {
+  const { t } = useTranslation();
+  const { setFieldValue } = useFormikContext<FormikValues>();
+  const isBuildV1Enabled = useFlag(FLAG_OPENSHIFT_BUILDCONFIG);
+  const isShipwrightEnabled = useFlag(FLAG_SHIPWRIGHT_BUILD);
+  const isPipelineEnabled = useFlag(FLAG_OPENSHIFT_PIPELINE);
+  const hasCreatePipelineAccess = usePipelineAccessReview();
+
+  const fieldName = 'build.option';
+
+  const selectInputOptions = React.useMemo(() => {
+    const options: SelectInputOption[] = [];
+
+    if (isBuildV1Enabled) {
+      options.push({
+        label: t(`devconsole~${ReadableBuildOptions[BuildOptions.BUILDS]}`),
+        value: BuildOptions.BUILDS,
+        description: t(
+          'devconsole~Builds are a core concept in OpenShift Container Platform. A build describes a process for transforming source code into a runnable image.',
+        ),
+      });
+    }
+
+    if (isShipwrightEnabled) {
+      options.push({
+        label: t(`devconsole~${ReadableBuildOptions[BuildOptions.SHIPWRIGHT]}`),
+        value: BuildOptions.SHIPWRIGHT,
+        description: t(
+          'devconsole~Shipwright is a Kubernetes-native framework for building container images. Shipwright is a collection of Kubernetes CRDs and controllers that provide a declarative API for image builds.',
+        ),
+      });
+    }
+
+    if (isPipelineEnabled && hasCreatePipelineAccess) {
+      options.push({
+        label: t(`devconsole~${ReadableBuildOptions[BuildOptions.PIPELINES]}`),
+        value: BuildOptions.PIPELINES,
```

Comment:
Translation is not happening. Getting below message in console log, PTAL.

`i18n.js:111 Missing i18n key "Pipelines" in namespace "public" and language "en." `

You have to provide the const values as below

```
// t('devconsole~Serverless Deployment')
  [Resources.KnativeService]: `devconsole~Serverless Deployment`,
```